doc/user/content/sql/create-sink/kafka.md

```diff
@@ -352,12 +352,29 @@ Consider using the Debezium envelope if:
 ### Automatic topic creation
 
 If the specified Kafka topic does not exist, Materialize will attempt to create
-it while processing the `CREATE SINK` statement. Materialize will configure the
-topic with the broker's default number of partitions and replication factor.
+it when the sink starts. Materialize will configure the topic with the broker's
+default number of partitions and replication factor.
 
-To customize the topic configuration, create the sink's topic outside of
+Similarly, if the connection's [progress topic](#exactly-once-processing) does
+not exist, Materialize will attempt to create it when the sink starts.
+Materialize will configure the progress topic with a single partition, the
+broker's default replication factor, and compaction enabled.
+
+To customize the topics' configurations, create the topics outside of
 Materialize with the desired configuration (e.g., using the [`kafka-topics.sh`]
-tool) before running `CREATE SINK`.
+tool) before running `CREATE SINK`. If you choose to do so, observe the
+following guidance :
+
+| Topic          | Configuration       | Guidance
+|----------------|---------------------|---------
+| Data topic     | Partition count     | Your choice, based on your performance and ordering requirements.
+| Data topic     | Replication factor  | Your choice, based on your durability requirements.
+| Data topic     | Compaction          | Your choice, based on your downstream applications' requirements. If using the [Upsert envelope](#upsert), enabling compaction is a natural choice.
+| Data topic     | Retention           | Your choice, based on your downstream applications' requirements.
+| Progress topic | Partition count     | **Required** to be set to 1. Using multiple partitions can cause Materialize to violate its [exactly-once guarantees](#exactly-once-processing).
+| Progress topic | Replication factor  | Choose based on your durability requirements.
+| Progress topic | Compaction          | We recommend enabling compaction to avoid accumulating unbounded state. Disabling compaction may cause performance issues, but will not cause correctness issues.
+| Progress topic | Retention           | **Required** to be disabled. Enabling retention can cause Materialize to violate its [exactly-once guarantees](#exactly-once-processing).
```

Comment:
```suggestion
| Data topic     | Partition count     | Your choice, based on your performance and ordering requirements.
| Data topic     | Replication factor  | Your choice, based on your durability requirements.
| Data topic     | Compaction          | Your choice, based on your downstream applications' requirements. If using the [Upsert envelope](#upsert), we recommend enabling compaction.
| Data topic     | Retention           | Your choice, based on your downstream applications' requirements.
| Progress topic | Partition count     | **Must be set to 1**. Using multiple partitions can cause Materialize to violate its [exactly-once guarantees](#exactly-once-processing).
| Progress topic | Replication factor  | Your choice, based on your durability requirements.
| Progress topic | Compaction          | We recommend enabling compaction to avoid accumulating unbounded state. Disabling compaction may cause performance issues, but will not cause correctness issues.
| Progress topic | Retention           | **Must be disabled**. Enabling retention can cause Materialize to violate its [exactly-once guarantees](#exactly-once-processing).
```